scripts/amalgamation.py

```diff
@@ -299,6 +299,7 @@ def generate_duckdb_hpp(header_file):
             hfile.write("#define DUCKDB_AMALGAMATION_EXTENDED 1\n")
         hfile.write("#define DUCKDB_SOURCE_ID \"%s\"\n" % git_commit_hash())
         hfile.write("#define DUCKDB_VERSION \"%s\"\n" % git_dev_version())
+        hfile.write("#define DUCKDB_DISTRIBUTION \"amalgamation\"\n")
```

Comment:
Minor, but maybe:
```suggestion
        hfile.write("#ifndef DUCKDB_DISTRIBUTION \"\n")
        hfile.write("#define DUCKDB_DISTRIBUTION \"amalgamation\"\n")
        hfile.write("#endif \"\n")
```
I have not tried to actually compile this, but the idea should be that starting from amalgamation you can still provide it externally.

Reply:
done; tested that an external constant can be provided to amalgamation through cmake parameters